crates/red_knot/src/types/infer.rs

```diff
@@ -145,6 +145,15 @@ fn infer_expr_type(db: &dyn SemanticDb, file_id: FileId, expr: &ast::Expr) -> Qu
     // TODO cache the resolution of the type on the node
     let symbols = symbol_table(db, file_id)?;
     match expr {
+        ast::Expr::NumberLiteral(ast::ExprNumberLiteral { value: v, .. }) => {
+            if let ast::Number::Int(n) = v {
+                // TODO support big int literals, or at least default to `builtins.int`
+                Ok(n.as_i64().map(Type::IntLiteral).unwrap_or(Type::Unknown))
+            } else {
+                // TODO builtins.float or builtins.complex
+                Ok(Type::Unknown)
+            }
```

Comment:
I would probably use a match here since we want to handle all branches in the future (and rename `v` to `value`...)
```suggestion
					match value {
							ast::Number::Int(int) => Ok(n.as_i64().map(Type::IntLiteral).unwrap_or(Type::Unknown)),	
							ast::Number::Float(_) | ast::Number::Complex(_) => {
                // TODO builtins.float or builtins.complex
                Ok(Type::Unknown)
              }
          }
 ```